### DISCUSSION TITLE
Use relative paths for includes?

### DISCUSSION THREAD

**Discussion by Please-just-dont** [stephenberry/glaze#1393] [Ideas]
What's your thought about relative includes? In my opinion instead of:

#include "glaze/beve.hpp"
#include "glaze/beve/beve_to_json.hpp"
#include "glaze/csv.hpp"

have:

#include "beve.hpp"
#include "beve/beve_to_json.hpp"
#include "csv.hpp"

Or so that it's obvious to the reader:

#include "./beve.hpp"
#include "./beve/beve_to_json.hpp"
#include "./csv.hpp"

Now I don't have to set the glaze folder as an include directory when I compile.

**Comment by stephenberry** [comment]
The use of "glaze/..." in the headers is intentional and necessary. It prevents clashes between headers that have the same name in different dependencies.

Consider we have a dependency that includes "foo/csv.hpp". If we only included "csv.hpp" then this header would collide with Glaze's header and we wouldn't be able to use both libraries in the same codebase.

This is a common practice and recommended for C++ libraries.

**Reply by Please-just-dont** [reply]
Hmm, true. No conflict for ./ though right

**Reply by stephenberry** [reply]
There is still a conflict for "./" if two libraries have the same include structure, which is quite common.

The alternative solution is to put glaze in all the headers: "glaze_csv.hpp rather than "glaze/csv.hpp". But, this has worse organization.
